### ISSUE TITLE
Autocomplete breaks after Where statement

### ISSUE THREAD

**Issue by ufuk-ergin-carbon** [dbeaver/dbeaver#20199]
### Description

There is still a bug in autocomplete. Steps to replicate:
1. Have two tables to join with aliases each. Fill the columns of interest in the Select clause.
2. When you come to WHERE clause, if you have the first condition such as bn.column_name the autocomplete works. 
3. But add another condition with AND and the autocomplete breaks for everywhere in the whole query.

### DBeaver Version

Version 23.1.0.202306041918

### Operating System

MacOS Ventura 13.3.1 (a)

### Database and driver

Snowflake
select current_version():
7.18.1

Driver version:
3.13.6

### Steps to reproduce

1. Have two tables to join with aliases each. Fill the columns of interest in the Select clause.
2. When you come to WHERE clause, if you have the first condition such as bn.column_name the autocomplete works. 
3. But add another condition with AND and the autocomplete breaks for everywhere in the whole query.

**Comment by E1izabeth** [comment]
Thank you for bug report.

**Comment by E1izabeth** [comment]
As a workaround you can uncheck this setting
![image](https://github.com/dbeaver/dbeaver/assets/28875055/6320339c-54ae-4a47-97ae-90859303c76b)

**Comment by E1izabeth** [comment]
@ufuk-ergin-carbon Unfortunately, I couldn't reproduce your issue
![image](https://github.com/dbeaver/dbeaver/assets/28875055/ffdac903-db67-434e-b605-581f490a0223)

Could you clarify what am I doing not as you do?

**Comment by ufuk-ergin-carbon** [comment]
Hello E1izabeth,
could you please try appending another `AND` statement after gr.ssn != 0 and use alias?

**Comment by E1izabeth** [comment]
Hmm.. Still works for me

![image](https://github.com/dbeaver/dbeaver/assets/28875055/3b408660-2e65-4b76-a043-ea8c64e28186)

![image](https://github.com/dbeaver/dbeaver/assets/28875055/f6e26ec5-1063-48e1-8165-f337160ac092)

**Comment by ufuk-ergin-carbon** [comment]
alias completion works for me too, the autocomplete of columns do not work after the alias. Could you please try that as well?

**Comment by E1izabeth** [comment]
I tried a lot, still can't reproduce. Could you give me an exact query example? It could be a query text and your cursor position where autocomplete doesn't work for you.

**Comment by ufuk-ergin-carbon** [comment]
`SELECT * FROM ANALYTICS_PROD_DB.`

In the above query, parser used to autocomplete/suggest with "ANALYTICS" after the period. In this specific example, I left the computer for half an hour and came back to desk and now it does not work. Same with the below query:

```
SELECT
	docs.FIRST_NAME || ' ' || docs.LAST_NAME AS full_name
	bn.CREATED_AT ,
	count(DISTINCT bn.ID) AS something,
	count(DISTINCT bn.BILLID) AS something2
FROM ANALYTICS_PROD_DB.ANALYTICS.FACT_BILLNOTES bn
INNER JOIN STG_DOCTORS docs ON bn.DOCTORID = docs.DOCTOR_ID 
WHERE bn.DOCTORID !='e0ec78fa'
	**AND bn.**
```

Also, the parser autocomplete is inconsistent throughout. I know it is hard to replicate because it does not break in a consistent manner :(

Could you try to leave your computer and then come back to check if the parser still works, for example? It is frustrating to have to quit DBeaver and then relaunch it to make the parser do it its job.

**Comment by E1izabeth** [comment]
Thank you for details explanation

**Comment by E1izabeth** [comment]
And does this (https://github.com/dbeaver/dbeaver/issues/20199#issuecomment-1576330767 ) helped? Could you try? It'll help to investigate the issue.

**Comment by E1izabeth** [comment]
Also could you try pressing Invalidate/Reconnect button when autocompletion is broken? Does it help?

**Comment by ufuk-ergin-carbon** [comment]
Unfortunately I did all those. I also noticed that if you write a CTE the autocomplete also breaks on the queries within a CTE. See screenshots.
<img width="833" alt="Screenshot 2023-06-08 at 17 16 58" src="https://github.com/dbeaver/dbeaver/assets/114302087/90496649-3521-4152-9dda-0d0e66138c43">
<img width="495" alt="Screenshot 2023-06-08 at 17 16 37" src="https://github.com/dbeaver/dbeaver/assets/114302087/17bba01a-dcfe-4a14-bc36-f51980d0fc35">

**Comment by E1izabeth** [comment]
It doesn't work with CTE at the moment. It's expected. So, your issue is about CTE?

**Comment by ufuk-ergin-carbon** [comment]
No, it's just something I additionally discovered. My main problem was that it breaks after like half an hour. I'll try to replicate it again and redo Invalidate/Reconnect.

Do you have plans to launch autocomplete on CTEs as well?

**Comment by E1izabeth** [comment]
Yes, we do.

It would be useful if you could also provide log files when you reproduce it. You can find instructions here https://dbeaver.com/docs/wiki/Log-files/

**Comment by ufuk-ergin-carbon** [comment]
Hi,
I just lost the autocomplete even when I did not let DBeaver sit for 10 minutes. I am attaching the log files in zip format and two screenshots.
![Screenshot 2023-06-12 at 13 29 05](https://github.com/dbeaver/dbeaver/assets/114302087/78651a07-a744-4dbd-889b-443c88d62f49)
![Screenshot 2023-06-12 at 13 29 16](https://github.com/dbeaver/dbeaver/assets/114302087/c463f174-1751-41a1-9e3f-7c77930a0bc0)
[metadata.zip](https://github.com/dbeaver/dbeaver/files/11721328/metadata.zip)

Edit: I think I found a way to reproduce this bug. Try to mimic the query in the screenshot. Run the command to get the counts.

Then, go back to the query and try to add another column with the table alias name. As you see, the autocomplete breaks.

**Comment by ufuk-ergin-carbon** [comment]
I think I found the problem. The problem is the GROUP BY statement. If you enter a space between FROM and GROUP BY thereby separating them, the addition of another column with table alias works. Let me know if you need more explanation.

**Comment by saesiva** [comment]
This is also happening to me. I noticed that if I erased all of the contect in the SELECT statement, the autocomplete started working again, and for me it mostly happens when I try to operate any of the fields with any function (sum, count, case etc), if I just call for the columns it mostly works (though it sometimes fails).

disabling the "Enable Experimental Features" seems to have done the trick for me at least for now, will follow up. Hope this helps.

**Comment by E1izabeth** [comment]
Could you check if the issue is still reproducible on the latest DBeaver version with "Enable experimental features" on?

**Comment by ufuk-ergin-carbon** [comment]
Hi, yes, I just tried and it fails to autosuggest. I turned it off and reconnected and it worked fine.

**Comment by E1izabeth** [comment]
Thank you for update

**Comment by ma3xdev** [comment]
Hello and sorry to open this issue again, but I think I found a specific bug related to autocomplete:
While I wrote the following query, autocomplete always worked great:
```sql
SELECT
	prod.codice AS "ID",
	mag.sku AS "EAN",
	upper(TRANSLATE(prod.codiceproduttore, '\/:*?<>|-_ .', '')) AS "CODICE BONIFICATO",
	marca.nomeit AS "MARCA",
	genere.nomeit AS "SESSO",
	cat.pathnomeit AS "CATEGORIA"
FROM prodotto AS prod
	INNER JOIN magazzino AS mag ON (mag.prodotto_id = prod.id)
	INNER JOIN marca ON (marca.id = prodotto.marca_id)
	INNER JOIN genere ON (genere.id = prodotto.genere_id)
	INNER JOIN categoria AS cat ON (cat.id = prod.categoria3_id)
WHERE prod.codiceproduttore <> ''
ORDER BY prod.id, mag.id
;
```
But, if I change the line with the translate, adding a double quote in the string to translate (which is what I'd like to have, as it is allowed in the SQL language), from
```sql
upper(TRANSLATE(prod.codiceproduttore, '\/:*?<>|-_ .', '')) AS "CODICE BONIFICATO",
```
to
```sql
upper(TRANSLATE(prod.codiceproduttore, '\/:*?"<>|-_ .', '')) AS "CODICE BONIFICATO",
```
then the autocomplete stops working from the line following that line.
![Screenshot 2024-03-25 at 15 29 26](https://github.com/dbeaver/dbeaver/assets/57186730/e6cf21bc-4dbf-4eb7-b6c7-7f0efa684dc3)

**Comment by E1izabeth** [comment]
Hello, @ma3xdev 
Thank you for reporting it. We're glad to improve our syntax highlighting. Please create a new issue for this specific case.

**Comment by ma3xdev** [comment]
Hello, @E1izabeth 

Done with the issue #23228 
Thank you for your fast reply.

**Comment by kestlr01** [comment]
Looks like the last comment on this was from Aug 23. We are experiencing the same issues as @ufuk-ergin-carbon. We are on the latest version of the community edition and we have started a trial of the lite edition as well.
